components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -590,11 +590,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             throw new ApplicationError(ErrorCodes.NOT_AUTHENTICATED, "User is not authenticated. Please login.");
         }
 
-        const user = await this.userDB.findUserById(this.userID);
-        if (!user) {
-            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "User does not exist.");
```

Comment:
it will be another exception now

Reply:
Yes, `userService.findUserB<ID` throws an ApplicationError(NOT_FOUND)

Reply:
I'm not sure what was intent to throw bad requests, maybe security reasons

Reply:
I'm pretty sure it was just a bad choice for this case